Google.PowerShell/Sql/GcSqlInstanceCmdlets.cs

```diff
@@ -513,18 +535,48 @@ private class ParameterSetNames
         [Parameter(Mandatory = false, Position = 4, ParameterSetName = ParameterSetNames.Csv)]
         public string[] Column { get; set; }
 
+        /// <summary>
+        /// <para type="description">
+        ///  If true, uploads the object at ImportFilePath to a new Google Cloud Storage bucket, then imports it
+        ///  into the instance. Otherwise, imports the file at the Google Cloud Storage location
+        ///  specified by ImportFilePath
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter UploadLocalFile { get; set; }
+
+        /// <summary>
+        /// MIME attachment for general binary data. (Octets of bits, commonly referred to as bytes.)
+        /// </summary>
+        protected const string OctetStreamMimeType = "application/octet-stream";
+
+        private string bucketName;
+        private StorageService bucketService;
```

Comment:
If you were to use the `GcsFileUploader` type described in the comment above, then in the implementation of your cmdlet you can just keep track of Bucket and Object fields as needed.

````
Bucket tempGcsBucket = null;
Object tempGcsObject = null;
GcsFileUploader tempUploader = null;

// ...

if (needToUploadLocalFiles) {
    tempFileUploader = new GcsFileUploader();
    tempGcsBucket = tempFileUploader.CreateBucket(...)
    tempGcsObject = tempFileUploader.CreateObject(tempGcsBucket, localFilePath);
}

// And then for cleanup.
if (tempGcsObject != null) {
    tempFileUploader.DeleteObject(tempGcsObject);
    tempFileUploader.DeleteBucket(tempGcsObject.Bucket);
}
````